src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
+        }
+        if (index == 0) {
+            node.prev = null;
+            node.next = head;
+            head.prev = node;
+            head = node;
+        } else {
+            Node<T> temp = getNode(index);
+            node.prev = temp.prev;
+            node.next = temp;
+            temp.prev.next = node;
+            temp.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).currentValue;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        T newValue = node.currentValue;
+        node.currentValue = value;
+        return newValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.currentValue;
+        unlink(node);
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (object == node.currentValue || object != null && object.equals(node.currentValue)) {
+                unlink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        Node<T> node = head;
+        if (index < size / 2) {
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = tail;
+            for (int i = size; i > index + 1; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index - " + index
+                    + " out of bounds for length: " + size);
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        if (node == head) {
+            if (size != 1) {
+                head.next.prev = null;
+                head = head.next;
+            } else {
+                head = null;
+                tail = null;
+            }
+            size--;
+            return;
+        }
+        if (node == tail) {
+            tail.prev.next = null;
+            tail = tail.prev;
+            size--;
+            return;
+        }
+        node.next.prev = node.prev;
+        node.prev.next = node.next;
+        size--;
```

Comment:
Looks like we can make an improvment here. You can `size--` first step in this method, to avoid duplications. And call `size != 0` or better `!isEmpty()` on the line 127